components/tel/Core/Src/main.c

```diff
@@ -0,0 +1,232 @@
+/* USER CODE BEGIN Header */
+/**
+  ******************************************************************************
+  * @file           : main.c
+  * @brief          : Main program body
+  ******************************************************************************
+  * @attention
+  *
+  * Copyright (c) 2022 STMicroelectronics.
+  * All rights reserved.
+  *
+  * This software is licensed under terms that can be found in the LICENSE file
+  * in the root directory of this software component.
+  * If no LICENSE file comes with this software, it is provided AS-IS.
+  *
+  ******************************************************************************
+  */
+/* USER CODE END Header */
+/* Includes ------------------------------------------------------------------*/
+#include "main.h"
+#include "cmsis_os.h"
+#include "can.h"
+#include "i2c.h"
+#include "usart.h"
+#include "gpio.h"
+
+/* Private includes ----------------------------------------------------------*/
+/* USER CODE BEGIN Includes */
+
+#include "can.h"
+
+/* USER CODE END Includes */
+
+/* Private typedef -----------------------------------------------------------*/
+/* USER CODE BEGIN PTD */
+
+/* USER CODE END PTD */
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+
+/* USER CODE BEGIN PV */
+
+CAN_FilterTypeDef CAN_filter0;
+CAN_FilterTypeDef CAN_filter1;
+
+CAN_RxHeaderTypeDef can_rx_header;
+
+HAL_StatusTypeDef can_start;
+
+uint8_t current_can_data[8];
+HAL_StatusTypeDef rx_status;
```

Comment:
This is pretty standard for our projects because we used superloops for the most part and then copied code over for subsequent projects. Since there is no longer CAN logic code in `main.c/h`, it would be nicer to encapsulate the CAN code all in `can.c` and `can.h` and then have a function call to setup anything necessary in main (which gets included from `can.h`). For example, after calling `MX_CAN_Init()` we call a single function `Can_Init()` which sets up the filters, starts CAN with HAL, and activates notifications for interrupts.

This just makes things easier to find and cleaner.